src/astarte-client/transforms/transformData.ts

```diff
@@ -0,0 +1,106 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2024 SECO Mind Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import {
+  AstarteInterfaceValues,
+  AstarteDatastreamIndividualData,
+  AstarteDatastreamObjectData,
+  AstarteDataValue,
+  AstarteDataTuple,
+  AstarteInterface,
+} from 'astarte-client';
+import _ from 'lodash';
+
+export const transformObjectData = (
+  fetchedData: AstarteInterfaceValues,
+  nesto: AstarteInterfaceValues,
+): AstarteDatastreamObjectData[] => {
+  const result: AstarteDatastreamObjectData[] = [];
+
+  const processNode = (node: any) => {
+    if (Array.isArray(node)) {
+      node.forEach((item) => processNode(item));
+    } else if (node && typeof node === 'object' && node.timestamp) {
+      const { timestamp, ...data } = node;
+      const path = Object.keys(nesto);
+      const transformedData: AstarteDatastreamObjectData = {
+        endpoint: `/${path}`,
+        timestamp,
+        value: data,
+      };
+      result.push(transformedData);
+    } else if (node && typeof node === 'object') {
+      Object.keys(node).forEach((subPath) => {
+        processNode(node[subPath]);
+      });
+    }
+  };
+
+  processNode(fetchedData);
+
+  return result;
+};
```

Comment:
1. What is the purpose of this function? What does it do? The code is not immediately clear.
2. Why two sets of `AstarteInterfaceValues` are needed as input?
3. What does `nesto` mean? We could use more meaningful, english names.